From 31374f27d482a1a201003f086f5041c60704efbd Mon Sep 17 00:00:00 2001
From: Mark Salyzyn <salyzyn@google.com>
Date: Wed, 29 Aug 2018 16:58:52 -0700
Subject: [PATCH 8/8] init: First Stage Mount observe nofail mount flag

MountPartitions should respect nofail.  The nofail flag makes the
partition optional.

Test: manual (future code using feature), and smoke test
Bug: 109821005
Bug: 78914864
Change-Id: I715a66ef1abbe223b0cefe5bd2758f63352b1afc
---
 init/init_first_stage.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init/init_first_stage.cpp b/init/init_first_stage.cpp
index dcec21de2..ad2791b6f 100644
--- a/init/init_first_stage.cpp
+++ b/init/init_first_stage.cpp
@@ -313,7 +313,7 @@ bool FirstStageMount::MountPartitions() {
     }
 
     for (auto fstab_rec : mount_fstab_recs_) {
-        if (!MountPartition(fstab_rec)) {
+        if (!MountPartition(fstab_rec) && !fs_mgr_is_nofail(fstab_rec)) {
             return false;
         }
     }
-- 
2.17.1

